File: semester_6_2023-2024/cme_3202_concepts_of_programming_languages/project.go
package main

import (
  "fmt"
  "math"
  "strconv"
  "strings"
)

func main() {
  var point1Str, point2Str string

  fmt.Println("Enter coordinates for first point (comma-separated): ")
  fmt.Scanln(&point1Str)  // Read entire line with Scanln

  fmt.Println("Enter coordinates for second point (comma-separated): ")
  fmt.Scanln(&point2Str)

  // Split strings and convert to float64 (inline)
  point1StrArr := strings.Split(point1Str, ",")
  point2StrArr := strings.Split(point2Str, ",")
  point1 := make([]float64, len(point1StrArr))
  point2 := make([]float64, len(point2StrArr))

  for i, val := range point1StrArr {
    val = strings.TrimSpace(val) // Remove leading/trailing whitespaces
    f, err := strconv.ParseFloat(val, 64)
    if err != nil {
      fmt.Println("Error parsing point1:", err)
      return
    }
    point1[i] = f
  }
  
  for i, val := range point2StrArr {
    val = strings.TrimSpace(val) // Remove leading/trailing whitespaces
    f, err := strconv.ParseFloat(val, 64)
    if err != nil {
      fmt.Println("Error parsing point2:", err)
      return
    }
    point2[i] = f
  }

  // Ensure both points have the same dimensionality
  if len(point1) != len(point2) {
    fmt.Println("Points must have the same number of dimensions.")
    return
  }

  var sumSqDiff float64
  // Calculate squared difference for each dimension (inline)
  for i := range point1 {
    diff := point2[i] - point1[i]
    sumSqDiff += math.Pow(diff, 2)
  }

  // Calculate and display distance (inline)
  distance := math.Sqrt(sumSqDiff)
  fmt.Printf("Euclidean distance between points: %.2f\n", distance)
}

File: semester_6_2023-2024/cme_3202_concepts_of_programming_languages/project.rb
puts "Enter coordinates of point 1 (comma separated):"
point1 = gets.chomp # Get input from user, remove newline character

puts "Enter coordinates of point 2 (comma separated):"
point2 = gets.chomp # Get input from user, remove newline character

# Split the input strings into arrays of floats
coordinates1 = point1.split(",").map(&:to_f) 
coordinates2 = point2.split(",").map(&:to_f)

# Ensure both points have the same dimensionality
if coordinates1.size != coordinates2.size
  raise ArgumentError, "Points must have the same number of dimensions"
end

# Calculate the squared difference for each dimension
squared_differences = coordinates1.zip(coordinates2).map { |x, y| (x - y) ** 2 }

# Sum the squared differences and take the square root
distance = Math.sqrt(squared_differences.sum)

# Display the distance
puts "Euclidean distance between points: #{distance}"